src/api.rs

```diff
@@ -34,12 +45,14 @@ pub fn register_player(
     // Broadcast to all hosts that a new player has joined.
     broadcaster.send(HostBroadcast::PlayerRegistered(PlayerData {
         id: player_id,
+        username: name_clone.clone(),
         score: 0,
     }));
 
     // Respond to the client.
     JSON(RegisterPlayerResponse {
         id: player_id,
+        username: name_clone,
```

Comment:
Since in the previous 2 places you use the username (adding it to the username table and broadcasting to the hosts) you cloned it, you can still use the original `username` variable:

```
username: username,
```